### PR TITLE
Replace prettytable with texttable

### PR DESCRIPTION
**Pull request by CyberDem0n**
It is a more advanced library and can deal with multi-line strings, what allows us to present tags in a nice format.
The redundant column `Column` will be presented in the table header.

Depending on output format `Tags` are serialized differently:
* For *pretty* format YAML is used, every element on the new line
* For *tsv* format for YAML is also used, but all elements and on the same line (similar to JSON)
* For *json* and *yaml* formats `Tags` are serialized into an appropriate format.

<details><summary>Examples of output in pretty formats:</summary>

```bash
$ patronictl list
┌ Cluster: batman (6792870884189948744) ┬─────────┬────┬───────────┬─────────────────┬────────────────────────────┐
│   Member    │      Host      │  Role  │  State  │ TL │ Lag in MB │ Pending restart │           Tags             │
├─────────────┼────────────────┼────────┼─────────┼────┼───────────┼─────────────────┼────────────────────────────┤
│ postgresql0 │ 127.0.0.1:5432 │ Leader │ running │  2 │           │                 │                            │
├─────────────┼────────────────┼────────┼─────────┼────┼───────────┼─────────────────┼────────────────────────────┤
│ postgresql1 │ 127.0.0.1:5433 │        │ running │  2 │         0 │                 │ clonefrom: true            │
│             │                │        │         │    │           │                 │ nofailover: true           │
│             │                │        │         │    │           │                 │ noloadbalance: true        │
│             │                │        │         │    │           │                 │ replicatefrom: postgresql0 │
├─────────────┼────────────────┼────────┼─────────┼────┼───────────┼─────────────────┼────────────────────────────┤
│ postgresql2 │ 127.0.0.1:5434 │        │ running │  2 │         0 │ *               │ replicatefrom: postgres1   │
└─────────────┴────────────────┴────────┴─────────┴────┴───────────┴─────────────────┴────────────────────────────┘

$ patronictl list badclustername
┌ Cluster: badclustername (uninitialized) ──────┐
│ Member │ Host │ Role │ State │ TL │ Lag in MB │
└────────┴──────┴──────┴───────┴────┴───────────┘

$ patronictl history
┌────┬──────────┬──────────────────────────────┬───────────────────────────┐
│ TL │   LSN    │            Reason            │         Timestamp         │
├────┼──────────┼──────────────────────────────┼───────────────────────────┤
│  1 │ 25657792 │ no recovery target specified │ 2020-02-13T11:26:27+01:00 │
│  2 │ 25690088 │ no recovery target specified │ 2020-02-13T11:29:53+01:00 │
└────┴──────────┴──────────────────────────────┴───────────────────────────┘

$ patronictl query -c "SELECT a, repeat('x', a) from generate_series(1,3) a"
┌───┬────────┐
│ a │ repeat │
├───┼────────┤
│ 1 │ x      │
│ 2 │ xx     │
│ 3 │ xxx    │
└───┴────────┘
```
</details>

<details><summary>Example in tsv format:</summary>

```bash
Cluster Member  Host    Role    State   TL      Lag in MB       Pending restart Tags
batman  postgresql0     127.0.0.1:5432  Leader  running 2
batman  postgresql1     127.0.0.1:5433          running 2       0               {clonefrom: true, nofailover: true, noloadbalance: true, replicatefrom: postgresql0}
batman  postgresql2     127.0.0.1:5434          running 2       0       *       {replicatefrom: postgres1}
```
</details>

In addition to that, `patronictl list` command will stop showing keys with empty values in `json` and `yaml` formats.
<details><summary>Examples:</summary>

#### YAML
```yaml
$ patronictl list -f yaml
- Cluster: batman
  Host: 127.0.0.1:5432
  Member: postgresql0
  Role: Leader
  State: running
  TL: 2
- Cluster: batman
  Host: 127.0.0.1:5433
  Lag in MB: 0
  Member: postgresql1
  State: running
  TL: 2
  Tags:
    clonefrom: true
    nofailover: true
    noloadbalance: true
    replicatefrom: postgresql0
- Cluster: batman
  Host: 127.0.0.1:5434
  Lag in MB: 0
  Member: postgresql2
  Pending restart: '*'
  State: running
  TL: 2
  Tags:
    replicatefrom: postgres1
```

### JSON
```json
$ patronictl list -f json | jq .
[
  {
    "Cluster": "batman",
    "Member": "postgresql0",
    "Host": "127.0.0.1:5432",
    "Role": "Leader",
    "State": "running",
    "TL": 2
  },
  {
    "Cluster": "batman",
    "Member": "postgresql1",
    "Host": "127.0.0.1:5433",
    "State": "running",
    "TL": 2,
    "Lag in MB": 0,
    "Tags": {
      "nofailover": true,
      "noloadbalance": true,
      "replicatefrom": "postgresql0",
      "clonefrom": true
    }
  },
  {
    "Cluster": "batman",
    "Member": "postgresql2",
    "Host": "127.0.0.1:5434",
    "State": "running",
    "TL": 2,
    "Lag in MB": 0,
    "Pending restart": "*",
    "Tags": {
      "replicatefrom": "postgres1"
    }
  }
]
```
</details>

P.S. Debian/Ubuntu already have the module packaged as `python3-texttable`, so hopefully it shouldn't become a big problem for package maintainers. CC: @mbanck

### CODE REVIEW

File: patroni/ctl.py [hunk 1]
```diff
@@ -133,31 +133,32 @@ def request_patroni(member, method='GET', endpoint=None, data=None):
...
+        list_cluster = bool(header and columns and columns[0] == 'Cluster')
+        if list_cluster and 'Tags' in columns:  # we want to format member tags as YAML
+            i = columns.index('Tags')
+            for row in rows:
+                if row[i]:
+                    row[i] = format_config_for_editing(row[i], fmt == 'tsv').strip()
+        s = int(list_cluster and fmt == 'pretty')  # skip cluster name if pretty-printing
+        table_data = ([columns[s:]] if columns else []) + [row[s:] for row in rows]
+
+        if fmt == 'tsv':
```
Comment (sdudoladov) [inline patroni/ctl.py]: `patronictl  list --help` does not mention the `tsv` format

**Comment by sdudoladov** [comment]
:+1:

**Comment by mbanck** [comment]
> P.S. Debian/Ubuntu already have the module packaged as `python3-terminaltables`, so hopefully it shouldn't become a big problem for package maintainers. CC: @mbanck

One possible problem with that is that `python3-terminaltables` only seems to be available from `buster` on, while we tried to also support older Debian releases via `apt.postgresql.org`. However, I had issues with the latest 1.6.4 packages on `stretch`, so maybe this point is moot. I'll try to look into this again today and tomorrow.

Or maybe it could just remove that output format if that module is not available? Didn't look at the code yet, maybe it already does.

**Comment by CyberDem0n** [comment]
My biggest issue with prettytables that it doesn't support multiline cells, therefore we can't nicely show tags.
Getting rid of pretty format isn't really an option.
We can think of using `tabulate` instead of `terminaltables`, it doesn't look so cool, but packages are available.

**Comment by caseyallenshobe** [comment]
Wouldn't this work even with prettytables, using 2 additional rows with null values for the first seven cells?

```
# patronictl list
+------------------------------+-----------------------+---------------+--------+---------+----+-----------+---------------+
|           Cluster            |         Member        |      Host     |  Role  |  State  | TL | Lag in MB |  Tags         |
+------------------------------+-----------------------+---------------+--------+---------+----+-----------+---------------+
| postgresql-caseydev1-12-main | d-gp2-dbpgcaseydev1-1 |  10.124.193.5 | Leader | running |  1 |           |               |
| postgresql-caseydev1-12-main | d-gp2-dbpgcaseydev1-2 |  10.124.193.6 |        | running |  1 |         0 |  nofailover   |
|                              |                       |               |        |         |    |           | noloadbalance |
|                              |                       |               |        |         |    |           |    nosync     |
| postgresql-caseydev1-12-main | d-gp2-dbpgcaseydev1-3 | 10.124.193.28 |        | running |  1 |         0 |               |
+------------------------------+-----------------------+---------------+--------+---------+----+-----------+---------------+
```

**Comment by CyberDem0n** [comment]
@mbanck, what is your opinion on texttable?
Package seems to be available for Debian stretch.
But at the same time it is not available on xenial... :(

**Pull request closed without merging**